### ISSUE TITLE
CRITICAL Linter finished without analyzing any file, check configuration and arguments given.

### ISSUE THREAD

**Issue by iafelix** [ansible/ansible-lint#3322]
##### Summary

After upgrade `ansible-lint` from `6.14.6` to `6.15.0` it started to fail when running against non-ansible files, for example executed by a GitHub workflow for all YAML files changed.

The error message is below, it also exit with return code 5:

```plain
Passed with production profile: 0 failure(s), 0 warning(s) on 0 files.
CRITICAL Linter finished without analyzing any file, check configuration and arguments given.
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```shell
$ ansible-lint --version
ansible-lint 6.15.0 using ansible 2.14.4

# MacOS X
$ uname -a
Darwin iafelix.local 22.4.0 Darwin Kernel Version 22.4.0

# Alpine container
$ uname -a
Linux 6f0ffe2f6a08 5.15.49-linuxkit #1 SMP Tue Sep 13 07:51:46 UTC 2022 x86_64 Linux
```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

Both were installed using `pip`.

##### STEPS TO REPRODUCE

The YAML files from `ansible-lint` repo can be used to reproduce the error.

```console
$ git clone git@github.com:ansible/ansible-lint.git

$ cd ansible-lint
```

Before upgrade

```console
$ pip install ansible-lint==6.14.6

$ ansible-lint --version
ansible-lint 6.14.6 using ansible 2.14.4
A new release of ansible-lint is available: 6.14.6 → 6.15.0

$ pip list | grep ansible
ansible             7.4.0
ansible-builder     1.2.0
ansible-core        2.14.4
ansible-lint        6.14.6

$ ansible-lint .github/ISSUE_TEMPLATE/*.yml

Passed with production profile: 0 failure(s), 0 warning(s) on 0 files.

$ echo $?
0
```

The upgrade

```console
$ pip uninstall ansible-lint

$ pip install ansible-lint
```

After upgrade

```console
$ ansible-lint --version
ansible-lint 6.15.0 using ansible 2.14.4

$ pip list | grep ansible
ansible             7.4.0
ansible-builder     1.2.0
ansible-core        2.14.4
ansible-lint        6.15.0

$ ansible-lint .github/ISSUE_TEMPLATE/*.yml
Passed with production profile: 0 failure(s), 0 warning(s) on 0 files.
CRITICAL Linter finished without analyzing any file, check configuration and arguments given.

$ echo $?
5
```

##### Desired Behavior

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
The linter returned exit code 5 because it did not analyze any files. You missed to include the most important part: which command line arguments are you passing to the linter and in which directory/project you are running it. You should use `-v` or even `--v` to get more information as this will likely help you realize why it did not look at any files.

Hint: if everything is excluded. You might also want to create an empty git repo, add the file to it and run the linter on that, just to check that the project director is correctly identified.

For example, i am aware of an existing bug when you pass a random folder as an argument, one that is from outside the detected project directory. If the project directory is incorectly detected, you might try to use `--project-dir`.

**Comment by iafelix** [comment]
No args were given, neither required to reproduce the issue. If you just copy and paste all commands that I sent, you will see that on previous version that didn't happen. 

Note that I just navigated to a random github repo to reproduce the issue. 

The expected will be exit code 0 so it won't fail the workflow if there are no Ansible Playbook or Roles to validate. 

Please also note how it behaves on the previous version. 

If nothing has changed, nothing changes and that's OK. Not an issue

**Comment by ssbarnea** [comment]
I will close it as not being a bug and just a desirable behavior, one that would prevent false-positive accidents where people believed linter run, but on zero files. If nothing is recognized as lintable, it will give an error.

As linter knows to analyze random YAML files too, it will count all of them.

**Comment by iafelix** [comment]
I disagree with the solution. 

This is a new behavior as it wasn't working like that before. If you like I can send you the very same tests with as many versions as you like before the latest one. 

I would suggest to add an option to at least add that rule to the skip list. Or maybe change the  error message level that linter display from critical  to a warning, and it should return a non-zero return code only if the --strict option is passed in. That's is an acceptable closure for this issue. 

Otherwise it's entirely up to user to decide if the linter must be called on the work flow. That's a lot of duplicated logic as the linter is capable of to know what yaml files should be analyzed or just skip them.

**Comment by iafelix** [comment]
Got it... You closed as you are the one that created this bug trying to fix misconfiguration which is not the case. You didn't fix misconfiguration and created a truly undesired behavior. 

https://github.com/ansible/ansible-lint/commit/80e2be37f9db80913fd4c79a9772e498e459065a

I don't understand why you are not willing to work on that and treated this as "working as designed" really disappointed.
